src/category_theory/limits/shapes/binary_products.lean

```diff
@@ -246,22 +246,85 @@ abbreviation coprod.map {W X Y Z : C} [has_colimits_of_shape.{v} (discrete walki
   (f : W ⟶ Y) (g : X ⟶ Z) : W ⨿ X ⟶ Y ⨿ Z :=
 colim.map (map_pair f g)
 
+section prod_lemmas
+variable [has_limits_of_shape.{v} (discrete walking_pair) C]
+
 @[reassoc]
-lemma prod.map_fst {W X Y Z : C} [has_limits_of_shape.{v} (discrete walking_pair) C]
+lemma prod.map_fst {W X Y Z : C}
   (f : W ⟶ Y) (g : X ⟶ Z) : prod.map f g ≫ prod.fst = prod.fst ≫ f := by simp
 
 @[reassoc]
-lemma prod.map_snd {W X Y Z : C} [has_limits_of_shape.{v} (discrete walking_pair) C]
+lemma prod.map_snd {W X Y Z : C}
   (f : W ⟶ Y) (g : X ⟶ Z) : prod.map f g ≫ prod.snd = prod.snd ≫ g := by simp
 
+@[simp] lemma prod_map_id_id {X Y : C} :
+  prod.map (𝟙 X) (𝟙 Y) = 𝟙 _ :=
+by tidy
+
+@[simp] lemma prod_lift_fst_snd {X Y : C} :
+  prod.lift prod.fst prod.snd = 𝟙 (X ⨯ Y) :=
+by tidy
+
+-- I don't think it's a good idea to make any of the following simp lemmas.
+@[reassoc]
+lemma prod_map_comm {A B X Y : C} (f : A ⟶ B) (g : X ⟶ Y) :
+  prod.map (𝟙 X) f ≫ prod.map g (𝟙 B) = prod.map g (𝟙 A) ≫ prod.map (𝟙 Y) f :=
+by tidy
+
+@[reassoc] lemma prod_functorial {X Y Z W : C} (f : X ⟶ Y) (g : Y ⟶ Z) :
```

Comment:
I don't like the name of this (and the following), suggestions more than welcome.

Reply:
`prod_map_comp_id`?

Reply:
And `prod_map_id_comp` for the other? That works, thanks.